src/main/java/gregtech/common/metatileentities/primitive/MetaTileEntityCharcoalPileIgniter.java

```diff
@@ -0,0 +1,432 @@
+                .aisle(" BBB ", "XCCCX", "XCCCX", "XCCCX", " DDD ")
+                .aisle(" BBB ", "XCCCX", "XCCCX", "XCCCX", " DSD ")
+                .aisle(" BBB ", "XCCCX", "XCCCX", "XCCCX", " DDD ")
+                .aisle("     ", " XXX ", " XXX ", " XXX ", "     ")
+                .where('S', MetaTileEntities.CHARCOAL_PILE_IGNITER, EnumFacing.NORTH)
+                .where('B', Blocks.BRICK_BLOCK.getDefaultState())
+                .where('X', Blocks.DIRT.getDefaultState())
+                .where('D', Blocks.GRASS.getDefaultState())
+                .where('C', Blocks.LOG.getDefaultState())
+                .build());
+    }
+
+    @Override
+    public NBTTagCompound writeToNBT(NBTTagCompound data) {
+        super.writeToNBT(data);
+        data.setInteger("lDist", this.lDist);
+        data.setInteger("rDist", this.rDist);
+        data.setInteger("hDist", this.hDist);
+        data.setInteger("progressTime", this.progressTime);
+        data.setInteger("maxProgress", this.maxProgress);
+        data.setBoolean("isActive", this.isActive);
+        return data;
+    }
+
+    @Override
+    public void readFromNBT(NBTTagCompound data) {
+        super.readFromNBT(data);
+        this.lDist = data.getInteger("lDist");
+        this.rDist = data.getInteger("rDist");
+        this.hDist = data.getInteger("hDist");
+        this.progressTime = data.getInteger("progressTime");
+        this.maxProgress = data.getInteger("maxProgress");
+        this.isActive = data.getBoolean("isActive");
+    }
+
+    @Override
+    public void writeInitialSyncData(PacketBuffer buf) {
+        super.writeInitialSyncData(buf);
+        buf.writeInt(this.lDist);
+        buf.writeInt(this.rDist);
+        buf.writeInt(this.hDist);
+        buf.writeInt(this.progressTime);
+        buf.writeInt(this.maxProgress);
+        buf.writeBoolean(this.isActive);
+    }
+
+    @Override
+    public void receiveInitialSyncData(PacketBuffer buf) {
+        super.receiveInitialSyncData(buf);
+        this.lDist = buf.readInt();
+        this.rDist = buf.readInt();
+        this.hDist = buf.readInt();
+        this.progressTime = buf.readInt();
+        this.maxProgress = buf.readInt();
+        this.isActive = buf.readBoolean();
+    }
+
+    @Override
+    public void receiveCustomData(int dataId, PacketBuffer buf) {
+        super.receiveCustomData(dataId, buf);
+        if (dataId == GregtechDataCodes.WORKABLE_ACTIVE) {
+            this.isActive = buf.readBoolean();
+            scheduleRenderUpdate();
+        }
+    }
+
+    /**
+     * Add a block to the valid Charcoal Pile valid wall/roof blocks
+     * @param block the block to add
+     */
+    @SuppressWarnings("unused")
+    public static void addWallBlock(@Nonnull Block block) {
+        WALL_BLOCKS.add(block);
+    }
+
+    @Override
+    public boolean isWorkingEnabled() {
+        return true;
+    }
+
+    @Override
+    public void setWorkingEnabled(boolean isActivationAllowed) {
+
+    }
```

Comment:
When you use soft mallet on this multiblock, it prints "Working Disabled" although it actually does not
you may want to redo soft mallet code to avoid misleading